src/Native/Unix/System.Native/pal_io.cpp

```diff
@@ -213,13 +213,13 @@ static int32_t ConvertOpenFlags(int32_t flags)
             ret = O_WRONLY;
             break;
         default:
-            assert(false && "Unknown Open access mode.");
+            assert_msg(false, "Unknown Open access mode", static_cast<int>(flags));
```

Comment:
This is passing flags as the error value; won't that be confusing, in particular when assert_msg prints out the strerror for it?